### PR TITLE
Fully migrate Fedora setup script to Ansible playbook

### PR DESCRIPTION
**Pull request by justwheel**
This pull request migrates my tedious Bash script for setting up a new Fedora host into an [Ansible playbook](https://www.ansible.com/). This playbook was successfully tested in a [Vagrant virtual machine](https://www.vagrantup.com/), but it was a headless environment. Recent additions came from the experience of reinstalling my desktop.

I'd like to open this PR up for comments from any friends or FOSS folk who are interested in Ansible automation on the desktop. Nitpicks welcome! :smile:

### CODE REVIEW

**Comment by linux-modder** [comment]
Comments:

- Leverage Jinja2 templates more for the config files and such
- add your user to docker (i.e usermod -aG) 
-  import ALL your gpg keys FIRST, especially with rpmfusion so you can load them and then install the bits needed.
- if a fileserver of any kind (even a local network nextcloud instance) use that for sanity checks and automated pulls, this also helps for otherwise slower connections and allows for less potential version bumping, alternatively, you could set the version locking on packages (even with state=latest)

File: ansible/setup_fedora.yml [hunk 1]
```diff
@@ -1,117 +1,191 @@
...
+        - bash_profile
+    - name: configure bash auto-completion scripts
+      file: force=yes src={{ git_conf_dir }}/taskwarrior/{{ item }} dest={{ user_home_dir }}/.bash_completion.d/{{ item }} state=link
+      with_items:
+        - task.sh
+    - name: configure bugwarrior
+      file: force=yes src={{ git_conf_dir }}/bugwarrior/bugwarriorrc dest={{ user_home_dir }}/.config/bugwarrior/bugwarriorrc state=link
+    - name: configure dunst
+      file: force=yes src={{ git_conf_dir }}/dunst/dunstrc dest={{ user_home_dir }}/.config/dunst/dunstrc state=link
+    - name: configue git
```
Comment (Tjzabel) [inline ansible/setup_fedora.yml]: s/configue/configure

**Comment by Tjzabel** [comment]
Looks good!

**Comment by justwheel** [comment]
Thanks for the feedback, @linux-modder. :+1: Some follow-up questions:

> Leverage Jinja2 templates more for the config files and such

Could you explain what you mean with this? I'm familiar with Jinja2, but I don't understand its application for desktop configuration files.

> add your user to docker (i.e usermod -aG)

I was doing this initially, but the playbook failed because the group was not created when the task ran. Do you know of a way to ensure the Docker daemon creates the group or can manually creating the group work here?

> import ALL your gpg keys FIRST, especially with rpmfusion so you can load them and then install the bits needed.

For the repositories where I did not import a signing key, it seems the certificate was provided or found during the setup process (since I didn't have any missing certificate errors). Is there a better reason to not let them be pulled in automatically?

File: ansible/setup_fedora.yml (hunk 1 above)
Comment (justwheel) [reply ansible/setup_fedora.yml]: @Tjzabel I can always count on you to catch my typos. :smile: Thanks!

**Comment by axk4545** [comment]
@jwflory honestly I might borrow some of this for my own playbook set up. I am curious about a few things though. 

- Rather than excluding packages from a GNOME install, why not do a net install of minimal and then install all the things you want?

- Why do you use `dnf` in some places and `package` in others?

- If you wanted to get rid of a lot of the `file` tasks that do the symlinking you could switch to using `yadm` to manage the dotfiles. It will handle the symlinks for you.

**Comment by axk4545** [comment]
@jwflory honestly I might borrow some of this for my own playbook set up. I am curious about a few things though. 

- Rather than excluding packages from a GNOME install, why not do a net install of minimal and then install all the things you want?

- Why do you use `dnf` in some places and `package` in others?

**Comment by justwheel** [comment]
Hey @axk4545, I hope you found some of these bits useful. :smile: 

> Rather than excluding packages from a GNOME install, why not do a net install of minimal and then install all the things you want?

I did this because I may not always have a net install or a fresh install – sometimes I may want to bring a workstation "into sync" if it has packages I don't want on it.

> Why do you use dnf in some places and package in others?

Thanks, I didn't notice that. Fixed in 980850d6305c76dd8b3730ee6ee6935801ecc182.

> If you wanted to get rid of a lot of the file tasks that do the symlinking you could switch to using yadm to manage the dotfiles. It will handle the symlinks for you.

I looked into `yadm` before, but it was an extra layer to work with and I might not always have it in every environment I'm working with. I'm okay with letting Ansible manage my symlinks for now, but thanks for the suggestion.

**Comment by justwheel** [comment]
I'm going to rebase this and merge. :clapper: Thanks everyone for the feedback!

**Pull request merged**